tower_cli/cli/transfer/receive.py

```diff
@@ -0,0 +1,119 @@
+import tower_cli
+from tower_cli.exceptions import TowerCLIError
+from tower_cli.cli.transfer import common
+from tower_cli.conf import settings
+from tower_cli.utils import parser
+import click
+
+class Receiver :
+
+    def receive(self, all=False, asset_input=None) :
+        # Extract and consolidate all of the items we got on the command line
+        assets_to_export = {}
+        has_assets = False
+        if all :
+            has_assets = True
+            for aType in common.SEND_ORDER :
+                if aType not in assets_to_export :
+                    assets_to_export[aType] = {'all': True, 'names': []}
+                assets_to_export[aType]['all'] = True
+        else :
+            for asset_type in asset_input:
+                assets_to_export[asset_type] = {'all': False, 'names': []}
+
+                for asset_name in asset_input[asset_type] :
+                    has_assets = True
+                    if asset_name == 'all' :
+                        assets_to_export[asset_type]['all'] = True
+                    else :
+                        assets_to_export[asset_type]['names'].append(asset_name)
+
+        if not has_assets :
+            raise TowerCLIError("Nothing was specified for export")
+
+        # These will be returned from this method
+        exported_objects = []
+
+        for type in assets_to_export :
+            # Load the API options for this type of asset
+            types_api_options = common.getAPIOptions(type)
+
+            # Now we are going to extract the objects from Tower and put them into an array for processing
+            acquired_assets_to_export = []
+            identifier = common.getIdentity(type)
+
+            # Now we are either going to get everything or just one item and append that to the assets_to_export
+            if assets_to_export[type]['all'] :
+                resources = tower_cli.get_resource(type).list()
```

Comment:
just auditing the list commands, does this need `all_pages=True`?

Reply:
Yes, it should, I'll get that fixed.